core/checksums/src/main/java/software/amazon/awssdk/checksums/SdkChecksum.java

```diff
@@ -39,7 +39,7 @@ public interface SdkChecksum extends Checksum {
     static SdkChecksum forAlgorithm(ChecksumAlgorithm algorithm) {
         switch (algorithm.algorithmId()) {
             case "CRC32C":
-                return Crc32cProvider.create();
+                return CrcChecksumProvider.create();
```

Comment:
I think this is confusing. Nothing about `CrcChecksumProvider.create()` says that we're returning a CRC32C checksum implementation. Can we make the name more descriptive?

Reply:
made it to crc32cImplementation